File: Sample-java/src/main/java/com/opera/ads/demo/fragment/BannerFragment.java
package com.opera.ads.demo.fragment;

import android.view.Gravity;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;

import com.opera.ads.AdError;
import com.opera.ads.AdSize;
import com.opera.ads.banner.BannerAd;
import com.opera.ads.banner.BannerAdListener;
import com.opera.ads.banner.BannerAdView;
import com.opera.ads.demo.R;

public class BannerFragment extends BaseFragment {

    private BannerAdView mBannerAdView;

    @Override
    protected boolean hasVideo() {
        return true;
    }

    @Override
    protected int getAdFormatStringId() {
        return R.string.banner_ad_format;
    }

    @Override
    protected void loadAd() {
        destroyAd();
        if (mBannerAdView == null) {
            mBannerAdView = new BannerAdView(getContext());
        }
        mLogView.print("Loading ...");
        final BannerAdListener adListener = new BannerAdListener() {
            @Override
            public void onAdLoaded(@NonNull BannerAd bannerAd) {
                mLogView.print("Loaded " + bannerAd.adSize + ", refreshCount: " + bannerAd.refreshCount);
                enableShowAd();
                enableDestroyAd();
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdError error) {
                mLogView.print("Error: " + error.getMessage());
            }

            @Override
            public void onAdImpression() {
                mLogView.print("Impression!");
            }

            @Override
            public void onAdClicked() {
                mLogView.print("Clicked!");
            }
        };
        mBannerAdView.setPlacementId(mPlacementId);
        mBannerAdView.setAdSize(AdSize.BANNER_MREC);
        mBannerAdView.loadAd(adListener);
    }

    @Override
    protected void showAd() {
        if (mBannerAdView != null) {
            if (mBannerAdView.isAdInvalidated()) {
                mLogView.print("Ad is invalidated");
                destroyAd();
                return;
            }
            FrameLayout.LayoutParams layoutParams = new FrameLayout.LayoutParams(
                    ViewGroup.LayoutParams.WRAP_CONTENT,
                    ViewGroup.LayoutParams.WRAP_CONTENT,
                    Gravity.CENTER);
            mAdContainer.addView(mBannerAdView, layoutParams);
            enableDestroyAd();
            disableShowAd();
        }
    }

    @Override
    protected void destroyAd() {
        super.destroyAd();
        if (mBannerAdView != null) mBannerAdView.destroy();
        mBannerAdView = null;
        mAdContainer.removeAllViews();
        disableShowAd();
        disableDestroyAd();
    }

    @Override
    public void onPause() {
        super.onPause();
        if (mBannerAdView != null) mBannerAdView.pause();
    }

    @Override
    public void onResume() {
        super.onResume();
        if (mBannerAdView != null) mBannerAdView.resume();
    }

    @Override
    public void onDestroy() {
        if (mBannerAdView != null) mBannerAdView.destroy();
        mBannerAdView = null;
        super.onDestroy();
    }
}

File: Sample-kotlin/src/main/kotlin/com/opera/ads/demo/fragment/BannerFragment.kt
package com.opera.ads.demo.fragment

import android.view.Gravity
import android.view.ViewGroup
import android.widget.FrameLayout
import com.opera.ads.AdError
import com.opera.ads.AdFormat
import com.opera.ads.AdSize
import com.opera.ads.banner.BannerAd
import com.opera.ads.banner.BannerAdListener
import com.opera.ads.banner.BannerAdView
import com.opera.ads.demo.R

class BannerFragment : BaseFragment(AdFormat.BANNER, true, true, R.string.banner_ad_format) {
    private var bannerAdView: BannerAdView? = null

    override fun loadAd() {
        destroyAd()
        if (bannerAdView == null) {
            bannerAdView = BannerAdView(requireContext())
        }
        bannerAdView?.run {
            logger.print("Loading ...")
            placementId = this@BannerFragment.placementId
            adSize = AdSize.BANNER_MREC
            loadAd(
                object : BannerAdListener {
                    override fun onAdLoaded(ad: BannerAd) {
                        logger.print("Loaded ${ad.adSize}, refreshCount: ${ad.refreshCount}")
                        enableShowAd()
                        enableDestroyAd()
                    }

                    override fun onAdFailedToLoad(error: AdError) {
                        logger.print("Error: ${error.message}")
                    }

                    override fun onAdImpression() {
                        logger.print("Impression!")
                    }

                    override fun onAdClicked() {
                        logger.print("Clicked!")
                    }
                }
            )
        }
    }

    override fun showAd() {
        bannerAdView?.run {
            if (isAdInvalidated()) {
                logger.print("Ad is invalidated")
                destroyAd()
                return
            }
            val layoutParams = FrameLayout.LayoutParams(
                ViewGroup.LayoutParams.WRAP_CONTENT,
                ViewGroup.LayoutParams.WRAP_CONTENT,
                Gravity.CENTER
            )
            adContainer().addView(this, layoutParams)
            enableDestroyAd()
            disableShowAd()
        }
    }

    override fun destroyAd() {
        bannerAdView?.destroy()
        bannerAdView = null
        adContainer().removeAllViews()
        logger.print("Destroyed...")
        disableShowAd()
        disableDestroyAd()
    }

    override fun onPause() {
        super.onPause()
        bannerAdView?.pause()
    }

    override fun onResume() {
        super.onResume()
        bannerAdView?.resume()
    }

    override fun onDestroy() {
        bannerAdView?.destroy()
        bannerAdView = null
        super.onDestroy()
    }
}
